api/src/services/collections.ts

```diff
@@ -98,6 +99,20 @@ export class CollectionsService {
 						};
 					}
 
+					// Add flags for specific database type overrides
+					switch (getDatabaseClient(this.knex)) {
+						case 'sqlite':
+							if (field.type === 'timestamp') {
+								addFieldFlag(field, 'cast-timestamp');
+							}
+							break;
+						case 'oracle':
+							if (field.type === 'dateTime') {
+								addFieldFlag(field, 'cast-datetime');
+							}
+							break;
+					}
```

Comment:
Can we abstract this away some more so we don't have to end up with "if db == 'x'" in more places in the codebase 🤔 

Maybe we can put it in one of those helpers, where it's like 

```js
addFieldFlag(helper.fieldFlagForField(field));
```

that way we don't have to do these explicit checks 🤔

Reply:
Great idea. Done. Thank you!